cmd/kube-apiserver/app/server.go

```diff
@@ -586,17 +585,6 @@ func BuildStorageFactory(s *options.ServerRunOptions) (*serverstorage.DefaultSto
 
 	// Set up encryption at rest via transformer overrides if a configuration file was provided.
 	if len(s.Etcd.EncryptionProviderConfigFilepath) != 0 {
-		cloud, err := cloudprovider.InitCloudProvider(s.CloudProvider.CloudProvider, s.CloudProvider.CloudConfigFile)
-		if err != nil {
-			return nil, err
-		}
-		// Returns an envelope Service implemented by a KMS supported by the cloud hosting k8s.
-		cloudKMSGetter := func(name string) (envelope.Service, error) {
-			return cloud.KeyManagementService(name)
-		}
-		// Register cloud in plugin registry in case a cloudprovided KMS provider is required.
-		encryptionconfig.KMSPluginRegistry.RegisterCloudProvidedKMSPlugin(cloudKMSGetter)
```

Comment:
Would it be acceptable if we move the registration of the KMS service provider to the main `init` function of the cloud? (The function which registers the cloudprovider itself). [This one](https://github.com/sakshamsharma/kubernetes/blob/89fd7cecf82adab67540be8237743d7becdcec99/pkg/cloudprovider/providers/gce/gce.go#L178)